4. Stack/priority_tasks/PriorityTasksArnab.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * PriorityTasksArnab.java
+ * Given an array of priority values,we have to check if the value next is of higher priority or not.If yes, then print that value,else
```

Comment:
In the problem description it say 'For each of the N tasks, your job is to find out the task with the next greater priority.' and in your description it say `Given an array of priority values ,we have to check if the value next is of higher priority or not.`. The problem is asking for the next greater element for an element, not if the next element is of higher priority, I believe your logic is flawed somewhere. 

Here is a test case:
1
6
3 2 6 7 1 2

Expected output: 6 6 7 -1 2 3
Your output: -1 6 7 -1 2 3

See if you can figure why this is happening.

Reply:
Yaa I understood the error, I am trying to resolve it